dev-guide/feature-gate.md

```diff
@@ -0,0 +1,280 @@
+---
+title: FeatureGates, round 2
+authors:
+- @deads2k
+reviewers:
+- team-leads # so they know how featuregates are changing.  The old way will work, but it will be unnecessarily hard.
+approvers:
+- @jspeed # api-approvers are picking up some code maintenance and we have to merge FeatureGate changes
+- staff-eng # so when asked by new team leads, they have some sense of how to do this.
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+- @jspeed
+creation-date: 2023-03-29
+last-updated: 2023-03-29
+---
+
+# FeatureGates, round 2
+
+## Summary
+
+This enhancement aims to reduce the effort required to add a feature gate to TechPreviewNoUpgrade and to promote
+that feature gate to Default.
+Feature gates in OpenShift are enabled and disabled in a particular FeatureSet, mixing and matching is not allowed.
+Prior to this enhancement, it is necessary to vendor openshift/api into every impacted repository.
+After this enhancement, it will only be necessary to vendor into cluster-config-operator.
+
+## Motivation
+
+Reduce the effort required to add and promote feature gates, to encourage and eventually require using
+feature gates to introduce new features into OCP.
+
+### User Stories
+
+As a staff engineer or release manager, I want to have no-developer-action result in features of unproven reliability
+inaccessible-by-default in long-lived clusters.
+
+As a developer, I want to have a low friction way to add a feature gate to TechPreviewNoUpgrade and to promote a feature
+gate to on-by-default.
+
+As an API approver, I want to have evidence of feature reliability before promoting feature gates to on-by-default. 
```

Comment:
it's a little weird to me that this implies it's an api approver decision.  I see that as a happenstance of where featuregates live, but i wouldn't expect api approvers to make the decision and i'd expect api approvers to respect it when it is made, not go examine the evidence and second guess it.  (that's as much about "they aren't the experts here" as it is "they don't have time to be doing that")

which begs an important question:  who *should* be making the decision?  
TRT?
a staff engineer?
the dev team?
QE?
some combination thereof?

Reply:
>  i wouldn't expect api approvers to make the decision and i'd expect api approvers to respect it when it is made

Similar to, "you need a merged enhancement" (api-approvers don't approve the enhancement), I would like to see, "you need reliability data" as a way to demonstrate, "yeah, basic threshold met so you can merge".  


> which begs an important question: who should be making the decision?

Do you see a reason to change the mechanics of how these are approved as part of this enhancement describing how they are consumed?  I'd rather debate mechanics of approval separately.

My final desire: bot does the merge by looking at the component reliability data.  My this year desire: staff eng decides by looking at the component reliability data.  My immediate goal: api-approvers aren't allowed to merge until some kind of feature reliability is presented.

Reply:
The mechanics of approval are my primary concern with this, so either we need to address that first or as part of this. We cannot create a system in which the forward progress of more than 370 people spread across all of the teams impacted by this change depends on the review capacity of 2 API reviewers or even 13 staff engineers.

Reply:
struck sentence. This is informal today and will not go away.  But changing it isn't a goal.  See pre-existing non-goal 2 on this list.